recipes/fmt/all/conanfile.py

```diff
@@ -81,6 +82,11 @@ def package_id(self):
     def validate(self):
         if self.settings.get_safe("compiler.cppstd"):
             check_min_cppstd(self, 11)
+        if self.settings.os == "Emscripten" and "20" in str(self.settings.compiler.cppstd):
```

Comment:
Not entirely sure these are worth it - I think invalidating configurations for the most "core" platforms where makes sense (e.g. if something doesnt build with gcc 11 with c++17) - but other things are diminishing returns. We'd probably have to stamp around half of recipes (if not more?) if we had to invalidate emscripten support. At _some_ point, I'd expect users to have some intuition that the problem is not the recipe and just the combination of upstream code and compiler/platform. 


On the other hand, I dont think the `Emscripten` OS is the problem - it's likely the version of LLVM/clang or libc++ that the emsdk is using? So there may be a newer version of emscripten with a compiler that doesn't have this problem - making this wrong. Also if the failure is a `consteval` if I understand correctly only has to do with the compiler.. unless there are emscripten macros involved.. dont think the OS is the problem, but the compiler version.

The issue referenced 4177 seems fixed via https://github.com/fmtlib/fmt/commit/cacc3108c5b74020dba7bf3c6d3a7e58cdc085b2 - has that not made it to a release we already have binaries for?

Reply:
Please, read https://github.com/conan-io/conan-center-index/pull/26177#issuecomment-2540839752